spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/web/ApplicationsController.java

```diff
@@ -60,18 +63,24 @@
                                                                   .map(tick -> PING);
     private final InstanceRegistry registry;
     private final InstanceEventPublisher eventPublisher;
+    private final Map<String, Application> requiredApplications;
 
-    public ApplicationsController(InstanceRegistry registry, InstanceEventPublisher eventPublisher) {
+    public ApplicationsController(InstanceRegistry registry, InstanceEventPublisher eventPublisher, String[] requiredApplications) {
         this.registry = registry;
         this.eventPublisher = eventPublisher;
+        this.requiredApplications = Stream.of(requiredApplications)
+                .map(ApplicationsController::requiredApplication)
+                .collect(toMap(Application::getName, Function.identity()));
     }
 
     @GetMapping(path = "/applications", produces = MediaType.APPLICATION_JSON_VALUE)
     public Flux<Application> applications() {
         return registry.getInstances()
                        .filter(Instance::isRegistered)
                        .groupBy(instance -> instance.getRegistration().getName())
-                       .flatMap(grouped -> toApplication(grouped.key(), grouped));
+                       .flatMap(grouped -> toApplication(grouped.key(), grouped))
+                       .concatWith(Flux.fromIterable(requiredApplications.values()))
+                       .distinct((app) -> app.getName());
```

Comment:
Here, I had no idea how to choose the 'required' application only if no other where present